src/components/fx/hover.js

```diff
@@ -1139,7 +1139,7 @@ function createHoverText(hoverData, opts) {
                 bgcolor: hoverlabel.bgcolor,
                 bordercolor: hoverlabel.bordercolor,
                 borderwidth: 1,
-                tracegroupgap: 7,
+                tracegroupgap: fullLayout.legend ? fullLayout.legend.tracegroupgap : undefined,
```

Comment:
Hmm I see we have the same logic on the next line for `traceorder` but this most likely predates multiple legends. I suppose this is going to give the correct behavior 99% of the time, but at some point we may want to look for which legend say the first trace shown in hover is included in, and use the `tracegroupgap` from that one.

Do we want to fall back on `undefined` or would the previous default of `7` be better if there's no legend at all? Or maybe change it to `10`, which is the default `tracegroupgap` when we DO have a legend, so adding an unstyled legend won't change the behavior here?

Reply:
@vladsavelyev any thoughts about @alexcjohnson 's comment? thanks - @gvwilson